src/main/java/co/cask/gcp/bigquery/BigQueryUtils.java

```diff
@@ -0,0 +1,247 @@
+ *  use this file except in compliance with the License. You may obtain a copy of
+ *  the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+
+package co.cask.gcp.bigquery;
+
+import co.cask.cdap.api.data.schema.Schema;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.hadoop.io.bigquery.BigQueryConfiguration;
+import com.google.common.collect.ImmutableMap;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.security.Credentials;
+import org.apache.hadoop.security.UserGroupInformation;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+import static co.cask.gcp.common.GCPUtils.loadServiceAccountCredentials;
+
+/**
+ * Common Util class for big query plugins such as {@link BigQuerySource} and {@link BigQuerySink}
+ */
+final class BigQueryUtils {
+  private static final Map<Schema.Type, LegacySQLTypeName> TYPE_MAP = ImmutableMap.<Schema.Type,
+    LegacySQLTypeName>builder()
+    .put(Schema.Type.INT, LegacySQLTypeName.INTEGER)
+    .put(Schema.Type.LONG, LegacySQLTypeName.INTEGER)
+    .put(Schema.Type.STRING, LegacySQLTypeName.STRING)
+    .put(Schema.Type.FLOAT, LegacySQLTypeName.FLOAT)
+    .put(Schema.Type.DOUBLE, LegacySQLTypeName.FLOAT)
+    .put(Schema.Type.BOOLEAN, LegacySQLTypeName.BOOLEAN)
+    .put(Schema.Type.BYTES, LegacySQLTypeName.BYTES)
+    .build();
+
+  private static final Map<Schema.LogicalType, LegacySQLTypeName> LOGICAL_TYPE_MAP = ImmutableMap.of(
+    Schema.LogicalType.DATE, LegacySQLTypeName.DATE,
+    Schema.LogicalType.TIME_MILLIS, LegacySQLTypeName.TIME, Schema.LogicalType.TIME_MICROS, LegacySQLTypeName.TIME,
+    Schema.LogicalType.TIMESTAMP_MILLIS, LegacySQLTypeName.TIMESTAMP,
+    Schema.LogicalType.TIMESTAMP_MICROS, LegacySQLTypeName.TIMESTAMP
+  );
+
+
+  /**
+   * Validates if schema of the provided field is simple type. This is because BigQuery plugins only support simple
+   * types. It retrieves non-nullable type from field, if field is of nullable type.
+   *
+   * @param field field for which schema needs to be validated
+   * @throws IllegalArgumentException if field type is not simple type
+   */
+  static void validateSimpleType(Schema.Field field) {
```

Comment:
this is a poor validation. Other types are not supported right? I thought floats and ints don't exist in BigQuery?

Reply:
Though it does make sense for the sink to support floats and ints, it doesn't make sense for the source to have them. This indicates that each plugin should validate differently. This is the type of static validation that should be in a validate() method of the corresponding plugin config